torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1839,8 +1848,10 @@ def hint():
                     source == symbol_to_source[expr][0]
                 ):
                     continue
-                sexpr = ShapeGuardPrinter(symbol_to_source, source_ref, self.var_to_sources).doprint(expr)
-                exprs.append(f"{source_ref(source)} == {sexpr}")
+                guard_sources = []
+                sexpr = ShapeGuardPrinter(symbol_to_source, source_ref, self.var_to_sources, guard_sources).doprint(expr)
+                guard_sources.append(source)
```

Comment:
This also seems weird

Reply:
Why don't you just make `source_ref` do the appending

Reply:
source_ref is passed in from above, so we would need to (1) intercede into source_ref with a wrapping fn (2) restart the guard_sources each time, felt not worth.

Reply:
But you had to modify at least two spots to manually append to sources. Modifying source_ref eliminates the manual lines of code and makes it obvious when you're supposed to append to sources.

Reply:
Well, if source_ref does the appending, you still have to pass a new list to it in 2 places? Or, alternatively, make it return a source alongside the ref, but then... you still need to append to build the ShapeGuardExprSources. 

I made the change, but I am not sold it adds value. Testing it now.

Reply:
Yeah, you ended up implementing this differently than how I was suggesting. Instead of padding it in as another argument to source_ref, you wrap source_ref inside symbolic_shapes to write to a sources_list variable that the function closes over. So it's super local to symbolic shapes (no need to change both places), no API change, but it's still obvious when to append.